### PR TITLE
[14.0][ADD] l10n_th_gov_purchase_guarantee

### PR DESCRIPTION
**Pull request by kittiu**
- [ ] Test

### CODE REVIEW

**Comment by kittiu** [comment]
@newtratip

**Comment by newtratip** [comment]
@kittiu, 

Could you check this commit https://github.com/OCA/l10n-thailand/pull/240/commits/ddeecaa49bd9f9b235bb47b238bde0c1af3aa075, I add doc type RFQ and PO in guarantee method for help filter guarantee in RFQ/PO

File: l10n_th_gov_purchase_guarantee/wizards/account_payment_register.py [hunk 1]
```diff
@@ -13,8 +13,8 @@ def _create_payments(self):
         active_model = self._context.get("active_model")
         active_ids = self._context.get("active_ids")  # must use active_ids
         if active_model == "account.move" and active_ids:
-            move = self.env[active_model].browse(active_ids)
-            move.ensure_one()
-            if move.return_guarantee_ids:
+            moves = self.env[active_model].browse(active_ids)
+            move_return_guarantee_ids = moves.filtered(lambda l: l.return_guarantee_ids)
+            for move in move_return_guarantee_ids:
                 move.return_guarantee_ids.date_return = self.payment_date
```
Comment (kittiu) [inline l10n_th_gov_purchase_guarantee/wizards/account_payment_register.py:19]: Can return_guarantee_ids be multi?

**Comment by kittiu** [comment]
@newtratip please make sure features are not removed. But rather, make it more generic.

File: l10n_th_gov_purchase_guarantee/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    {field: line[field] for field in list(line._fields.keys())}
+                )
+                new_line["move_id"] = rec.id
+                new_line._onchange_price_subtotal()
+                new_lines += new_line
+            new_lines._onchange_mark_recompute_taxes()
+            rec._onchange_currency()
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
```
Comment (kittiu) [inline l10n_th_gov_purchase_guarantee/models/account_move.py:71]: Please call super().

Note: This is a very dangerous code. I found the bug already created by this.

**Comment by kittiu** [comment]
> @newtratip please make sure features are not removed. But rather, make it more generic.

@newtratip please write note, what you have removed, and what will be returned. Thanks.

File: l10n_th_gov_purchase_guarantee/wizards/account_payment_register.py (hunk 1 above)
Comment (newtratip) [reply l10n_th_gov_purchase_guarantee/wizards/account_payment_register.py:19]: What do you mean ?

File: l10n_th_gov_purchase_guarantee/models/account_move.py (hunk 2 above)
Comment (newtratip) [reply l10n_th_gov_purchase_guarantee/models/account_move.py:71]: Fixed.

**Comment by newtratip** [comment]
I have return code back and add some features
- Guarantee Method can default method for Purchase Agreement, Request for Quotation, **Purchase Order** (Purchase Order is added)
- Add electronic guarantee type

**Comment by Saran440** [comment]
@kittiu this module is already merge. run on production.

**Review by Saran440 — APPROVED** [review]

**Review by newtratip — APPROVED** [review]
:+1:

**Pull request merged**